var/spack/repos/builtin/packages/py-pandas/package.py

```diff
@@ -129,6 +131,19 @@ class PyPandas(PythonPackage):
     # Optional dependencies
     # https://pandas.pydata.org/pandas-docs/stable/getting_started/install.html#optional-dependencies
 
+    # Excel dependencies for 1.4+ (not coded up for earlier versions)
```

Comment:
Missing odfpy.

It's easier to review if you keep the deps in the same order as in `pyproject.toml` and sort newest to oldest just like we do for versions.

Reply:
I was going off of https://pandas.pydata.org/docs/getting_started/install.html#install-source, where odfpy is listed in "Other data sources", but not in the section "Excel files". I added the package (it wasn't in spack yet) and also verified its dependencies being there. I also rearranged the dependencies as you suggested, and tested installing py-pandas@1.5.3 (succeeded) and py-pandas@2.0.3 (did not succeed on my macOS, because py-numba requires the real LLVM compiler and not just apple-clang, and the install of llvm@14 failed - didn't investigate further, because we only need py-pandas@1.5.3 on the macs for now; but all the dependencies that I configured for variant excel built fine).

Noting that py-numba is a "recommended dependency" and that it might be ok to not have it on macOS when using apple-clang?

Reply:
LLVM issues should be fixed by #40191